cerebrium/fine-tuning/auto-deploying.mdx

```diff
@@ -0,0 +1,51 @@
+---
+title: Auto-Deploying your Trained Model
+description: Learn how to seamlessly deploy your trained model to a production environment.
+---
+
+# Introduction
+
+If you are reading this guide, chances are that you have successfully trained a model on Cerebrium. Congratulations!
+
+Now that your model has been successfully trained, you have two options for deploying to a production environment:
+
+- The first is to download the trained adapter weights using `cerebrium download-model` and write your own deployment code.
+- The second is to use the built-in auto-deployment feature of Cerebrium's trainer.
+
+While you have more flexibility writing custom deployment code for your finetuned weights, the second option is the method that we recommend for most users!  
+It's faster, easier and gets your model up and running in no time.
+
+Auto-deployment streamlines the process of getting your model into a production environment. With this approach, deployment occurs automatically as soon as training is completed. This feature not only saves time but also provides a seamless way to test your model's performance before considering any modifications to the deployment code.
+
+An auto-deployment can be configured in the `config.yml` file. The auto-deployment will be triggered as soon as the training is complete.
+
+# Configuration of your Auto-deployment
+
+While an auto-deployment will generate all the code needed to deploy your model, you'll need to configure a few parameters to get it up and running. These parameters are added to the `config.yml` file of your training under the title `autodeploy`.
+
+We've kept the configuration as simple as possible, using the same format and parameters that you're used to in the [config file](cerebrium/cortex/advanced-functionality/init-cortex-project) of a Cortex deployment.
+
+The following parameters are the parameters we suggest you supply for an auto-deployment:
+
+```yaml
+autodeploy:
+  name: your-auto-deployed-model-name
+  hardware: AMPERE_A5000 # Hardware to use for the deployment
+  cpu: 2 # Number of CPUs to allocate to the deployment
+  memory: 14.5 # Memory to allocate to the deployment. This depends on your model but for most 7B models, 14.5GB is sufficient.
+  autodeploy-requirements-file: /Path/To/Your/requirements.txt # Requirements that you would need in a cortex deployment
```

Comment:
I don't understand why these are included if we set the main.py? They wont know what libraries we have included

Reply:
Imports are generated from the requirements.txt 
It's more there if a user wants a specific version of a package. 

But yes, at the moment they won't know what libraries we've included.